Chore: Add CloudWatch HTTP API tests

aknuds1 opened the pull request:
**What this PR does / why we need it**:
Add HTTP API tests for querying CloudWatch metrics and logs. Basically my proposed template for our future suite of HTTP API tests, hitting a real server instance (although in-process).

In order to enable API testing, I had to do some refactoring of the server logic, since the tests have to bring up a server instance (listening on a random port). It was also necessary to rewrite a bit so a per-server database engine would be used, instead of a global one via the bus. Should be a step in the right direction anyway, right?

Putting the PR up for feedback, although I want to wait for Carl and Marcus to be back from vacation so they can also provide feedback on this proposal for API testing.

In reviewing this, I recommend starting with pkg/server/api_metrics_test.go, since that contains the test itself and everything follows from that.

If desirable, I can divide the PR into smaller ones, although seeing the changes in their entirety is helpful since they're done in order to enable API testing.

@papagian Here are functioning API tests if you need something to base your own tests on.

patstrom committed:
cloudwatch: Explicitly specify internal client interface

This commit explicitly specifies and internal interface for the
cloudwatch executor to fetch the different AWS clients.

Previously each kind of client has been implemented in a different way
which was especially painful when implementing tests. With the commit
there is an explicit way to handle creation, fetching and testing of AWS
clients and interactions.

This lays the groundwork for reworking the caches in the cloudwatch
implementation altogether. The idea is to cache everything on a data
source id basis rather than the current cache keys, which are somewhat
arbitrary.

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-interface

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
CloudWatch: Use utility methods to get clients

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
CloudWatch: Refactor

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
CloudWatch: Rewrite tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Add test

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Add test

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Add test

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Remove unused code

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Clean up

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-interface

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-interface

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-interface

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-interface

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-interface

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-interface

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-interface

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-interface

aknuds1 committed:
Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Implement metrics query tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Implement regions query tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Implement instance attribute query tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Implement ebs volume query tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Implement resource ARN query tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Rename variable

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Restore go.sum

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-interface

aknuds1 committed:
CloudWatch: Clean up code

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
CloudWatch: Clean up code

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
CloudWatch: Clean up code

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge branch 'chore/cloudwatch-cleanup' into chore/cloudwatch-interface

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Clean up

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge branch 'chore/cloudwatch-cleanup' into chore/cloudwatch-interface

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge branch 'chore/cloudwatch-cleanup' into chore/cloudwatch-interface

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Unexport DatasourceInfo

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Unexport DatasourceInfo

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Unexport symbol

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Unexport symbol

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge branch 'chore/cloudwatch-cleanup' into chore/cloudwatch-interface

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Clean up

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge branch 'chore/cloudwatch-cleanup' into chore/cloudwatch-interface

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Rename variable

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge branch 'chore/cloudwatch-cleanup' into chore/cloudwatch-interface

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-interface

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-interface

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-interface

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-interface

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-interface

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-interface

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-interface

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-interface

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-interface

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-interface

aknuds1 committed:
API: Add test for getting CloudWatch metrics

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-interface

aknuds1 committed:
Server: Support initializing HTTP server with listener

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Server: Implement metrics API test

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-interface

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

papagian commented on pkg/tsdb/cloudwatch/metric_find_query.go:
```diff
@@ -713,6 +713,7 @@ func (e *cloudWatchExecutor) getAllMetrics(region string) (cloudwatch.ListMetric
 
 	plog.Debug("Listing metrics pages")
 	var resp cloudwatch.ListMetricsOutput
+	plog.Debug("Listing metrics pages")
```
This debug message is the same as 2 lines above.

aknuds1 replied:
@papagian Ooops, thanks, it's left over from a merge.

papagian commented on pkg/server/testdata/test.ini:
```diff
@@ -0,0 +1,761 @@
+# possible values : production, development
```
I wonder if this is necessary since this file is generated [here](https://github.com/grafana/grafana/pull/26517/files#diff-6fbf282916b625fa025c46f40dd231d2R105)

aknuds1 replied:
@papagian Oops, no it shouldn't be there. Thanks!

papagian reviewed (changes requested):
Nice work, I have some suggestions/questions.

aknuds1 committed:
Apply suggestions from code review

Co-authored-by: Sofia Papagiannaki <papagian@users.noreply.github.com>

aknuds1 committed:
fs: Switch out path for filepath

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Remove stale stuff

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 commented on pkg/server/api_metrics_test.go:
```diff
@@ -0,0 +1,267 @@
+	"xorm.io/xorm"
+)
+
+func createGrafDir(t *testing.T) (string, string) {
```
@papagian Yeah, I expect there will be more files, but my thinking is it makes sense to refactor when adding more tests, since it's only then you will know what's needed. See the YAGNI (You Ain't Gonna Need It) principle :)

I'm also wondering if the test should be renamed a bit, but I'm unsure to what. I named it in order to correlate with the tested API endpoint, and lack familiarity with the semantics of the endpoint. Do you know if the semantics of this endpoint is to _only_ query CloudWatch? I can't really tell.

aknuds1 replied:
I renamed the test to `TestQueryCloudWatchMetrics`, since at least for now it's 100% for CloudWatch.

papagian replied:
> Do you know if the semantics of this endpoint is to only query CloudWatch? I can't really tell.

no it's not only for cloudwatch. So the aim is to test the endpoint not. specifically cloudwatch, ok then.

aknuds1 replied:
@papagian It depends a bit on how we want to do it. Maybe there should be separate tests for different uses of the endpoint, or maybe they should be sub-tests. Not sure.

aknuds1 committed:
Rename test

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
server: Remove unused field

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
HTTPServer: Don't dependency inject route register

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
server: Add CloudWatch Logs API test

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Rename test helper function

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Remove stale comment

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Add more logging

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Modify CI tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Add debug logic

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
server: Combine CloudWatch API tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Move test helper functions

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Revert Drone config

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Clean up

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Split tests into two

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Fix querying

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

bergquist commented on pkg/api/datasources.go:
```diff
@@ -286,7 +286,7 @@ func GetDataSourceIdByName(c *models.ReqContext) Response {
 func (hs *HTTPServer) CallDatasourceResource(c *models.ReqContext) {
 	datasourceID := c.ParamsInt64(":id")
-	ds, err := hs.DatasourceCache.GetDatasource(datasourceID, c.SignedInUser, c.SkipCache)
+	ds, err := hs.DatasourceCache.GetDatasource(datasourceID, c.SignedInUser, c.SkipCache, nil)
```
Passing `nil` like that seems less desirable. Can we think of a better way?

aknuds1 replied:
@bergquist I can try to come up with something.

bergquist reviewed (commented):
Sorry for being slow on this PR..

I think this capability makes sense. I really like that we test the routing of the server. But I would love it if this could be turned into multiple PR. Too much going on to get a good picture I think.

aknuds1 commented:
@bergquist I can turn it into multiple PRs, no problem. Agree it's too large for a detailed review.

aknuds1 commented on pkg/api/api.go:
```diff
@@ -435,5 +435,10 @@ func (hs *HTTPServer) registerRoutes() {
 	r.Delete("/api/snapshots/:key", reqEditorRole, Wrap(DeleteDashboardSnapshot))
 
+	// Health check
+	r.Get("/healthz", func(c *models.ReqContext) {
```
@bergquist Thanks, I will look at it.

aknuds1 replied:
Switched to using @bergquist's suggestion.

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
API tests: Use existing health check

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Remove unnecessary import

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Move SQL store to datasources.CacheService

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

marefr commented:
Need to resolve some conflicts before I'll review this. Not sure you're aware, but we have a couple of api tests already written in typescript, see https://github.com/grafana/grafana/tree/master/devenv/e2e-api-tests. I agree that testing by hitting a real server instance in general has it's advantages, but I'm not sure what the advantages are for core data sources where the queries  mainly are created and executed from the UI/frontend. To me it would make more sense with E2E driven from the UI to test core backend plugins. But I definitely see advantages for other types of API endpoints that interacts with the database.

aknuds1 commented:
@marefr The PR is under reworking, since it was split into smaller PRs on @bergquist's request. @bergquist has already said he agrees with the premise of the tests, but he wanted the PR divided up (already done), and some details done differently.

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

torkelo committed:
Automation: Update backport github action trigger (#28352)

It seems like GitHub has solved the problem of running github actions on PRs from forks with access to secrets. 

https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/#improvements-for-public-repository-forks

If I change the event that triggers it to pull_request_target the action is run in the context of the base instead of the merged PR branch

torkelo committed:
ColorSchemes: Adds more color schemes and text colors that depend on the background (#28305)

* Adding more color modes and text colors that depend on the background color

* Updates

* Updated

* Another big value fix

* Fixing unit tests

* Updated

* Updated test

* Update

* Updated

* Updated

* Updated

* Updated

* Added new demo dashboard

* Updated

* updated

* Updated

* Updateed

* added beta notice

* Fixed e2e test

seanluce committed:
AzureMonitor: Fix capitalization of NetApp 'volumes' namespace (#28369)

'Microsoft.NetApp/netAppAccounts/capacityPools/Volumes'  -> 'Microsoft.NetApp/netAppAccounts/capacityPools/volumes'

bergquist committed:
Instrumentation: Adds environment_info metric  (#28355)

Signed-off-by: bergquist <carl.bergquist@gmail.com>

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

dprokop committed:
Explore: Minor tweaks to exemplars marble (#28366)

aknuds1 committed:
Plugins: Compose filesystem paths with filepath.Join (#28375)

* plugins: Fix filesystem path composition

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* plugins: Use filepath.Join to join filesystem paths

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
CloudWatch: Fix API tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Refactor middleware into service

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Convert from using global settings

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Port middleware logger changes from master

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Data source proxy: Rewrite tests to drop Convey

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Clean up

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Clean up

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/cloudwatch-tests
